src/applications/virtual-agent/tests/components/webchat/startConvoAndTrackUtterances.unit.spec.js

```diff
@@ -0,0 +1,199 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import StartConvoAndTrackUtterances from '../../../components/webchat/startConvoAndTrackUtterances';
+
+describe('makeBotStartConvoAndTrackUtterances actions', () => {
+  // mock store
+  const middlewares = [thunk];
+  const mockStore = configureMockStore(middlewares);
+  let fakeNext;
+  let store;
+  const connectFulfilledAction = { type: 'DIRECT_LINE/CONNECT_FULFILLED' };
+  const connectSendMessage = {
+    type: 'WEB_CHAT/SEND_MESSAGE',
+    payload: { text: 'some@email.com' },
+  };
+  const directIncomingActivity = {
+    type: 'DIRECT_LINE/INCOMING_ACTIVITY',
+    payload: { activity: 'some activity' },
+  };
+
+  beforeEach(() => {
+    fakeNext = sinon.stub();
```

Comment:
Very cool that we can mock the curried dependencies. I like this test setup where we can just focus on testing the incoming actions.